### PR TITLE
+cls #16050 Support graceful shutdown of ShardRegion

### PR DESCRIPTION
**Pull request by patriknw**
- possibility to define handOffStopMessage that is used
  for stopping the entries, both when graceful shutdown
  and rebalance

### CODE REVIEW

**Comment by patriknw** [comment]
Refs #16050

File: akka-docs/rst/scala/cluster-sharding.rst [hunk 1]
```diff
@@ -274,6 +275,15 @@ using a ``Passivate``.
...
 e.g. with ``akka-persistence``.
 
+Graceful Shutdown
+-----------------
+
+You can send the message ``ClusterSharding.GracefulShutdown`` message (``ClusterSharding.gracefulShutdownInstance
+in Java) to the ``ShardRegion`` actor to handoff all shards that are hosted by that ``ShardRegion`` and then the
+``ShardRegion`` actor will be stopped. You can ``watch`` the ``ShardRegion`` actor to know when it is completed.
+During this period other regions will buffer messages for those shards in the same way as when a rebalance is
+triggered by the coordinator. When the shards have been stopped the coordinator will allocate these shards elsewhere.
```
Comment (anderius) [inline akka-docs/rst/scala/cluster-sharding.rst]: If the coordinator lives on the same node, should this actor also be stopped gracefully?

Comment (patriknw) [inline akka-docs/rst/scala/cluster-sharding.rst]: I don't think that matters. You would watch the ShardRegion actor, and when it terminates you would cluster.leave(cluster.selfAddress) and that will trigger the graceful failover of the coordinator singleton.

This scenario should probably be explained here (with code).

**Comment by dragisak** [comment]
:+1:

**Comment by patriknw** [comment]
any reviewers from @akka/akka-team here (8 days old)?

**Comment by drewhk** [comment]
LGTM

**Comment by patriknw** [comment]
@anderius I have added the code example as I mentioned earlier, see 11bfe0f

**Comment by patriknw** [comment]
I would like to merge this after validation, so I can continue with other things in same area.

**Comment by patriknw** [comment]
@anderius merged. It will be in tomorrows timestamped snapshot if you want to try it out. Watch here: http://repo.akka.io/snapshots/com/typesafe/akka/akka-cluster-sharding_2.11/

**Pull request merged**